pkg/reconciler/pipelinerun/metrics_test.go

```diff
@@ -0,0 +1,144 @@
+package pipelinerun
+
+import (
+	alpha1 "github.com/tektoncd/pipeline/pkg/client/informers/externalversions/pipeline/v1alpha1"
+	"testing"
+	"time"
+
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	fakepipelineruninformer "github.com/tektoncd/pipeline/pkg/client/injection/informers/pipeline/v1alpha1/pipelinerun/fake"
+	tb "github.com/tektoncd/pipeline/test/builder"
+	corev1 "k8s.io/api/core/v1"
+	"knative.dev/pkg/apis"
+	"knative.dev/pkg/metrics/metricstest"
+	rtesting "knative.dev/pkg/reconciler/testing"
+)
+
+func Test_uninitialized_metrics(t *testing.T) {
+	metrics := Recorder{}
+
+	durationCountError := metrics.DurationAndCount(&v1alpha1.PipelineRun{})
+	prCountError := metrics.RunningPrsCount(nil)
+
+	assertErrNotNil(durationCountError, "DurationAndCount recording expected to return error but got nil", t)
+	assertErrNotNil(prCountError, "Current PR count recording expected to return error but got nil", t)
+}
+
+func Test_record_pipelinerun_duration_count(t *testing.T) {
+	startTime := time.Now()
+
+	testData := []struct {
+		name             string
+		taskRun          *v1alpha1.PipelineRun
+		expectedTags     map[string]string
+		expectedDuration float64
+		expectedCount    int64
+	}{
```

Comment:
style nit: I tend to prefer:

```
testData := []struct {

}{{
  ...
}, {
  ...
}
```

This preserves some vertical space, and quite a lot of horizontal space, which makes it easier to read I find.

Reply:
:+1: